tests/testutil/service.go

```diff
@@ -4,23 +4,54 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"os"
 	"testing"
 
 	riov1 "github.com/rancher/rio/pkg/apis/rio.cattle.io/v1"
 )
 
 type TestService struct {
-	Name    string // includes namespace and version
-	AppName string // name only
-	App     riov1.App
-	Service riov1.Service
-	Version string
-	T       *testing.T
+	Name       string // includes namespace and version
+	AppName    string // name only
+	App        riov1.App
+	Service    riov1.Service
+	Version    string
+	T          *testing.T
+	Kubeconfig string
 }
 
 // Create generates a new rio service, named randomly in the testing namespace, and
 // returns a new TestService with it attached. Guarantees ready state but not live endpoint
-func (ts *TestService) Create(t *testing.T, source string) {
+func (ts *TestService) Create(t *testing.T, source ...string) {
+	args := ts.createArgs(t, source...)
+	var envs []string
+	if ts.Kubeconfig != "" {
+		envs = []string{fmt.Sprintf("KUBECONFIG=%s", ts.Kubeconfig)}
+	}
+	_, err := RioCmd("run", args, envs...)
+	if err != nil {
+		ts.T.Fatalf("Failed to create service:  %v", err.Error())
+	}
+	err = ts.waitForReadyService()
+	if err != nil {
+		ts.T.Fatalf(err.Error())
+	}
+}
+
+func (ts *TestService) CreateWithError(t *testing.T, source ...string) error {
```

Comment:
Can you add a comment here just saying that this duplicates `Create` except that it returns an error.